### DISCUSSION TITLE
[Feature] Concurrent mobile upload

### DISCUSSION THREAD

**Discussion by aszdeff** [immich-app/immich#9710] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi,
in me installing and get immich working i need to backup all my pictures.
however the app only upload 1 picture every time. and it has a delay of 2-4sec between uploading another one. which for big uploads takes time. this is really a concern for initial uploaders.

could a  _**concurrent upload feature**_ like on the web ui be added on the app? 
(either a prepackaging of multiple pictures into one file then sent or just multiple uploads running simultaneously )

thanks for reading this :)

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by davidraedev** [comment]
I second this. I haven't had any time to look into why there is so much delay (at least thats what is implied in the app UI), but it seems like a perfect use case for concurrent uploads.
Even uploading just 20-30 pictures could see a big speedup.

**Comment by ghost** [comment]
+1

When initially setting up immich and syncing form phone parallel upload would be a great help

**Comment by LaVINCENTgit** [comment]
Hi all,
this problem still exist. When I try to upload photos from android chrome browser, just one photo or video is uploaded. 
(Why i use browser? Because in the mobile app is not possible to select in witch album upload contents).

Thanks

**Comment by TropicLegend** [comment]
Hello everybody! I second this. I have set up Immich and the initial upload of my phone's images could be sped up a lot with this feature.

Thank you and cheers!

**Comment by ProlinkX** [comment]
+1

**Comment by Anindya-Prithvi** [comment]
is anyone working on this? any branch i can have a look at? I can maybe try my hand here -

**Reply by alextran1502** [reply]
Already implemented in the beta timeline

**Reply by manalshaikh** [reply]
Hi. Is it released in prod? Because I'm facing this issue. Currently using Android app downloaded from appstore. It's latest.
